docroot/modules/custom/dcc_gtd_registration/src/Form/GlobalTrainingRegistrationForm.php

```diff
@@ -0,0 +1,466 @@
+<?php
+
+namespace Drupal\dcc_gtd_registration\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\PrependCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Entity\Entity;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Locale\CountryManager;
+use Drupal\Core\Render\Element\StatusMessages;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class GlobalTrainingRegistrationForm
+ *
+ * @package Drupal\dcc_gtd_registration\Form
+ */
+class GlobalTrainingRegistrationForm extends FormBase {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\dcc_gtd_registration\Form\CountryManager
+   */
+  protected $countryManager;
+
+  /**
+   * GlobalTrainingRegistrationForm constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   * The entity type manager.
+   * @param \Drupal\Core\Locale\CountryManager $countryManager
+   * The country manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, CountryManager $countryManager) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->countryManager = $countryManager;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public static function create(ContainerInterface $containerInterface) {
+    return new static(
+      $containerInterface->get('entity_type.manager'), $containerInterface->get('country_manager')
+    );
+  }
+
+  /**
+   * @return string
+   *
+   * Return the form Id
+   */
+  public function getFormId() {
+    return 'dcc_gtd_registration';
+  }
+
+  /**
+   * @param array $form
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *
+   * Creates the form elements for the first step
```

Comment:
Always keep the method comment on the first line and the param/return at the end.